python/tests/integration/arcticdb/test_arctic.py

```diff
@@ -45,40 +45,50 @@ class ParameterDisplayStatus(Enum):
 
 parameter_display_status = [ParameterDisplayStatus.NOT_SHOW, ParameterDisplayStatus.DISABLE, ParameterDisplayStatus.ENABLE]
 
-@S3_SSL_TESTS_MARK
+@pytest.mark.parametrize('storages, delimiter', [('s3_storage', '&'), pytest.param('s3_no_ssl_storage', '&', marks=SSL_TESTS_MARK), 
+                                                 pytest.param('azurite_storage', ';', marks=AZURE_TESTS_MARK), 
+                                                 pytest.param('azurite_ssl_storage', ';', marks=[AZURE_TESTS_MARK, SSL_TESTS_MARK])])
 @pytest.mark.parametrize('client_cert_file', parameter_display_status)
-@pytest.mark.parametrize('ssl', parameter_display_status)
-def test_s3_no_ssl_verification(s3_no_ssl_storage, client_cert_file, ssl):
-    uri = s3_no_ssl_storage.arctic_uri
-    if ssl == ParameterDisplayStatus.DISABLE:
-        uri += "&ssl=False"
-    elif ssl == ParameterDisplayStatus.ENABLE:
-        uri += "&ssl=True"
+@pytest.mark.parametrize('client_cert_dir', parameter_display_status)
+@pytest.mark.parametrize('ssl_setting', parameter_display_status)
+def test_s3_ssl_verification(request, monkeypatch, storages, delimiter, client_cert_file, client_cert_dir, ssl_setting):
```

Comment:
The parameterization and amount of logic in this test to combine the parameters in to a URI is really confusing. You should be able to clean this up a by just crafting the cases that you're interested in testing and passing a fully set URI in.

Have a look at `pytest.fixtures(params=...)`

The red flag is that you're parameterizing the test to run over the cartesian product of the fixtures, but then excluding a bunch of those cases by hand.

Reply:
I don't think that you've read my comment properly. I agree that we want the test cases, my issue is with how you are constructing them.